tests/wasm/runtime/entities.ts

```diff
@@ -40,26 +44,35 @@ export function fromData<D extends Definition>(definition: D, data: Data<D>): En
   return { entries }
 }
 
-function isRecursive(kind: RecursiveValueKind): kind is readonly [RecursiveValueKind] {
+function isArray(kind: EntityValueKind): kind is readonly [SimpleValueKind] {
   return Array.isArray(kind)
 }
 
-function coerceFromValue(value: Value, kind: RecursiveValueKind): unknown {
-  const unexpectedKindError = (expectedKind: ValueKind) => {
-    const n = (kind: ValueKind) => ValueKind[kind] || 'unknown'
-    return new Error(`expected ${n(expectedKind)} store value but got ${n(value.kind)}`)
-  }
+function isOptional(kind: EntityValueKind): kind is { readonly optional: SimpleValueKind } {
+  return typeof kind === 'object' && 'optional' in kind
+}
 
-  if (isRecursive(kind)) {
+function coerceFromValue(value: Value, kind: EntityValueKind): unknown {
+  if (isArray(kind)) {
     if (value.kind !== ValueKind.Array) {
-      throw unexpectedKindError(ValueKind.Array)
+      throw unexpectedKindError(value, ValueKind.Array)
```

Comment:
nit: should this check be part of the isArray method?

Reply:
This is checking that the specified `value` is an array value. The `isArray` just returns true if the `EthereumValueKind` is an array value.